code/game/machinery/teleporter.dm

```diff
@@ -55,202 +64,238 @@
 
 /obj/machinery/computer/teleporter/emag_act(mob/user)
 	if(!emagged)
-		emagged = 1
+		emagged = TRUE
 		to_chat(user, "<span class='notice'>The teleporter can now lock on to Syndicate beacons!</span>")
 	else
-		ui_interact(user)
+		tgui_interact(user)
 
 /obj/machinery/computer/teleporter/attack_ai(mob/user)
-	src.attack_hand(user)
+	attack_hand(user)
 
 /obj/machinery/computer/teleporter/attack_hand(mob/user)
-	ui_interact(user)
+	tgui_interact(user)
+
 
-/obj/machinery/computer/teleporter/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1)
+/obj/machinery/computer/teleporter/tgui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = TRUE, datum/tgui/master_ui = null, datum/tgui_state/state = GLOB.tgui_default_state)
 	if(stat & (NOPOWER|BROKEN))
 		return
-
-	// Set up the Nano UI
-	ui = SSnanoui.try_update_ui(user, src, ui_key, ui, force_open)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
 	if(!ui)
-		ui = new(user, src, ui_key, "teleporter_console.tmpl", "Teleporter Console UI", 400, 400)
+		ui = new(user, src, ui_key, "Teleporter", "Teleporter Console", 380, 260)
 		ui.open()
 
-/obj/machinery/computer/teleporter/ui_data(mob/user, ui_key = "main", datum/topic_state/state = GLOB.default_state)
-	var/data[0]
+/obj/machinery/computer/teleporter/tgui_data(mob/user)
+	var/list/data = list()
 	data["powerstation"] = power_station
 	if(power_station?.teleporter_hub)
 		data["teleporterhub"] = power_station.teleporter_hub
 		data["calibrated"] = power_station.teleporter_hub.calibrated
-		data["accurate"] = power_station.teleporter_hub.accurate
 	else
 		data["teleporterhub"] = null
 		data["calibrated"] = null
-		data["accurate"] = null
-	data["regime"] = regime_set
+	data["regime"] = regime
 	var/area/targetarea = get_area(target)
 	data["target"] = (!target || !targetarea) ? "None" : sanitize(targetarea.name)
 	data["calibrating"] = calibrating
-	data["locked"] = locked
+	data["locked"] = locked ? TRUE : FALSE
+	data["targetsTeleport"] = target_list
 	return data
 
-/obj/machinery/computer/teleporter/Topic(href, href_list)
+/obj/machinery/computer/teleporter/tgui_act(action, params)
 	if(..())
-		return 1
-
-	if(href_list["eject"])
-		eject()
-		SSnanoui.update_uis(src)
 		return
 
 	if(!check_hub_connection())
-		to_chat(usr, "<span class='warning'>Error: Unable to detect hub.</span>")
-		SSnanoui.update_uis(src)
+		atom_say("<span class='warning'>Error: Unable to detect hub.</span>")
```

Comment:
Dont use `atom_say()` and classes. One or the other.

Reply:
done